src/delegate.ts

```diff
@@ -0,0 +1,70 @@
+import { Handle } from 'dojo-core/interfaces';
+import on, { ExtensionEvent } from 'dojo-core/on';
+import has from './has';
+
+const matchMethod = has('element-match');
+
+/**
+ * Provides a normalized mechanism for using a single event handler to listen
+ * to delegated events from DOM nodes.
+ *
+ * @param target The Element to which to attach a single event handler
+ * @param selector A CSS selector used to determine whether the event handler is called
+ * @param type Event type(s) to listen for; may be strings or extension events
+ * @param listener Callback to handle the event when it fires
+ * @return A handle which will remove the listener when destroy is called
+ *
+ * @example
+ * dom.delegate(document.body, 'li', 'click', function () {
+ *     // ...
+ * });
+ *
+ * @example
+ * dom.delegate(document.body, 'li', ['click', 'mouseover'], function () {
+ *     // ...
+ * });
+ */
+export default function delegate(target: HTMLElement, selector: string, type: string, listener: (event: UIEvent) => void): Handle;
+export default function delegate(target: HTMLElement, selector: string, type: ExtensionEvent, listener: (event: UIEvent) => void): Handle;
+export default function delegate(target: HTMLElement, selector: string, type: (string | ExtensionEvent)[], listener: (event: UIEvent) => void): Handle;
+export default function delegate(target: HTMLElement, selector: string, type: any, listener: (event: UIEvent) => void): Handle {
+	function matches(node: HTMLElement, selector: string) {
```

Comment:
I am not sure that this function even needs to exist.  Is this a remnant of an earlier approach?  AFAICT directly calling the matches method down below where this is currently called works fine.  (I've changed it locally and the unit tests still pass.)